src/Toml/Codec/Types.hs

```diff
@@ -135,3 +133,79 @@ infixl 3 <!>
 (<!>) :: Alternative f => (a -> f x) -> (a -> f x) -> (a -> f x)
 f <!> g = \a -> f a <|> g a
 {-# INLINE (<!>) #-}
+
+{- | Mutable context for TOML conversion.
```

Comment:
Should we write that this monad is equivalent to `@MaybeT (State TOML)@`? Maybe it will be helpful for someone who knows monad transformers...